documentation/src/main/asciidoc/configuration.asciidoc

```diff
@@ -237,3 +237,54 @@ for beans injected into Hibernate Search.
 Beans <<configuration-bean-resolution,resolved by Hibernate Search>>
 using a <<configuration-bean-frameworks,supported framework>>
 can take advantage of injection features of this framework.
+
+[[configuration-background-failure-handling]]
+== Background failure handling
+// Search 5 anchors backward compatibility
+[[exception-handling]]
+
+Hibernate Search generally propagates exceptions occurring in background threads to the user thread,
+but in some cases, such as Lucene segment merging failures,
+the exception in background threads cannot be propagated.
+By default, when that happens, the failure is logged at the `ERROR` level.
+
+To customize background failure handling, you will need to:
+
+* Define a class that implements the `org.hibernate.search.engine.reporting.FailureHandler` interface.
+* Configure the backend to use that implementation by setting the configuration property
+`hibernate.search.background_failure_handler`
+to a <<configuration-property-types,bean reference>> pointing to the implementation.
```

Comment:
Are we sure that `Type of configuration properties` is the right chapter to link here?

Reply:
Yes, it explains what a bean reference is exactly.